File: legacy/src/main/java/br/com/legacy/utils/LongVersionSignature.java
package br.com.legacy.utils;

import com.bumptech.glide.load.Key;

import java.nio.ByteBuffer;
import java.security.MessageDigest;

/**
 * Created by appsimples on 11/7/17.
 */

public class LongVersionSignature implements Key {

    private long currentVersion;

    public LongVersionSignature(Long timestamp) {
        this.currentVersion = timestamp;
    }

    @Override
    public boolean equals(Object o) {
        if (o instanceof LongVersionSignature) {
            LongVersionSignature other = (LongVersionSignature) o;
            return currentVersion == other.currentVersion;
        }
        return false;
    }

    @Override
    public int hashCode() {
        return Long.valueOf(currentVersion).hashCode();
    }

    @Override
    public void updateDiskCacheKey(MessageDigest md) {
        md.update(ByteBuffer.allocate(Long.SIZE).putLong(this.currentVersion).array());
    }
}

File: component_user_session/src/main/kotlin/br/com/goin/component/session/user/signature/LongVersionSignature.kt
package br.com.goin.component.session.user.signature

import com.bumptech.glide.load.Key

import java.nio.ByteBuffer
import java.security.MessageDigest

class LongVersionSignature(private val currentVersion: Long) : Key {

    override fun equals(o: Any?): Boolean {
        if (o is LongVersionSignature) {
            val other = o as LongVersionSignature?
            return currentVersion == other!!.currentVersion
        }
        return false
    }

    override fun hashCode(): Int {
        return java.lang.Long.valueOf(currentVersion).hashCode()
    }

    override fun updateDiskCacheKey(md: MessageDigest) {
        md.update(ByteBuffer.allocate(java.lang.Long.SIZE).putLong(this.currentVersion).array())
    }
}
